mambaSharedFramework/InterstitialTagBuilder.swift

```diff
@@ -0,0 +1,293 @@
+//
+//  InterstitialTagBuilder.swift
+//  mamba
+//
+//  Created by Migneco, Ray on 10/22/24.
+//  Copyright © 2024 Comcast Corporation.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License. All rights reserved.
+//
+
+import Foundation
+
+
+
+/// A utility class for configuring and constructing an interstitial tag
+/// The properties in this class are in accordance with the HLS spec
+/// outlined in `draft-pantos-hls-rfc8216bis-15` Appendix D
+/// https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#appendix-D
+public final class InterstitialTagBuilder {
+    
+    /// An Interstitial EXT-X-DATERANGE tag MUST have a CLASS attribute whose
+    /// value is "com.apple.hls.interstitial".
+    static let appleHLSInterstitialClassIdentifier = "com.apple.hls.interstitial"
+    
+    /// A quoted-string that uniquely identifies a Date Range in the
+    /// Playlist.  This attribute is REQUIRED
+    let id: String
+    
+    /// required to be "com.apple.hls.interstitial"
+    let classId: String
+    
+    /// date/time at which the Date Range begins.  This attribute is REQUIRED.
+    let startDate: Date
+    
+    /// The value of the X-ASSET-URI is a quoted-string absolute URI for a
+    /// single interstitial asset.  An Interstitial EXT-X-DATERANGE tag
+    /// MUST have either the X-ASSET-URI attribute or the X-ASSET-LIST
+    /// attribute.  It MUST NOT have both.
+    let assetUri: String?
+    
+    /// The value of the X-ASSET-LIST is a quoted-string URI to a JSON
+    /// object.
+    let assetList: String?
+    
+    /// the duration of the interstitial content in seconds
+    var duration: Double?
```

Comment:
It is also useful to include `plannedDuration`. This helps AVPlayer know what range an interstitial _could_ occupy when the daterange `DURATION` is unknown without having to resolve the asset list/uri. This is particularly useful for ad insertions on the live edge, where the `PLANNED-DURATION` for the break is known, but we don't want to set the `DURATION` yet because there may be an early return to action where the break would need to be cut short.

```suggestion
    var duration: Double?
    
    /// the expected duration of the interstitial content in seconds which can indicate a value when the actual duration is not yet known
    var plannedDuration: Double?
```